### PR TITLE
Remove PHP Deprecated: Implicitly marking parameter as nullable.

### PR DESCRIPTION
**Pull request by terabytesoftw**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌

### CODE REVIEW

**Comment by terabytesoftw** [comment]
> Add info about refactoring to changelog

done.

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 - New #132: Add interface `SerializerInterface` for data serialization, and `PhpSerializer` implementation (@Gerych1984)
 - Chg #139: Make `normalize()` method static in `CacheKeyNormalizer` class (@terabytesoftw)
+- Chg #142: Remove PHP Deprecated: Implicitly marking parameter as nullable in PHP 8.4 (@terabytesoftw)
```
Comment (vjik) [inline CHANGELOG.md:7]:
```suggestion
- Enh #142: Minor refactoring: explicitly mark parameters as nullable (@terabytesoftw)
```

**Review by vjik — APPROVED** [review]

File: src/Cache.php [hunk 2]
```diff
@@ -63,7 +63,7 @@ public function getOrSet(
         mixed $key,
         callable $callable,
         DateInterval|int|null $ttl = null,
-        Dependency $dependency = null,
+        Dependency|null $dependency = null,
```
Comment (devanych) [inline src/Cache.php:66]: Maybe `?Dependency $dependency = null`?

File: src/CacheInterface.php [hunk 3]
```diff
@@ -59,7 +59,7 @@ public function getOrSet(
         mixed $key,
         callable $callable,
         DateInterval|int|null $ttl = null,
-        Dependency $dependency = null,
+        Dependency|null $dependency = null,
```
Comment (devanych) [inline src/CacheInterface.php:62]: Maybe `?Dependency $dependency = null`?

File: src/Exception/CacheException.php [hunk 4]
```diff
@@ -13,7 +13,7 @@ public function __construct(
         private string $key,
         string $message = '',
         int $code = 0,
-        Throwable $previous = null
+        Throwable|null $previous = null
```
Comment (devanych) [inline src/Exception/CacheException.php:16]: Maybe `?Throwable $previous = null`?

File: src/Dependency/TagDependency.php [hunk 5]
```diff
@@ -47,7 +47,7 @@ final class TagDependency extends Dependency
      * For a single tag, you may specify it as a string.
      * @param int|null $ttl The TTL value of this item. null means infinity.
      */
-    public function __construct(array|string $tags, int $ttl = null)
+    public function __construct(array|string $tags, int|null $ttl = null)
```
Comment (devanych) [inline src/Dependency/TagDependency.php:50]: Maybe `public function __construct(array|string $tags, ?int $ttl = null)`?

File: src/Cache.php (hunk 2 above)
Comment (terabytesoftw) [reply src/Cache.php:66]: I kept it with the same style of the package code, which uses type union, which is explicit.

File: src/CacheInterface.php (hunk 3 above)
Comment (terabytesoftw) [reply src/CacheInterface.php:62]: I kept it with the same style of the package code, which uses type union, which is explicit.

File: src/Exception/CacheException.php (hunk 4 above)
Comment (terabytesoftw) [reply src/Exception/CacheException.php:16]: I kept it with the same style of the package code, which uses type union, which is explicit.

File: src/Dependency/TagDependency.php (hunk 5 above)
Comment (terabytesoftw) [reply src/Dependency/TagDependency.php:50]: I kept it with the same style of the package code, which uses type union, which is explicit.

**Review by devanych — APPROVED** [review]

**Pull request merged**